actors/miner/src/lib.rs

```diff
@@ -4734,6 +4467,386 @@ fn balance_invariants_broken(e: Error) -> ActorError {
     )
 }
 
+fn extend_proof_validity<BS, RT>(
+    rt: &RT,
+    _: &ValidatedExpirationExtension,
+    sector: &SectorOnChainInfo,
+) -> Result<SectorOnChainInfo, ActorError>
+where
+    BS: Blockstore,
+    RT: Runtime<BS>,
+{
+    if !can_extend_seal_proof_type(sector.seal_proof) {
+        return Err(actor_error!(
+            forbidden,
+            "cannot extend expiration for sector {} with unsupported \
+                                    seal type {:?}",
+            sector.sector_number,
+            sector.seal_proof
+        ));
+    }
+    let policy = rt.policy();
+    let new_expiration =
+        sector.proof_expiration + (policy.max_proof_validity - policy.proof_refresh_window);
+    if new_expiration > rt.curr_epoch() + policy.max_proof_validity {
+        return Err(actor_error!(
+            forbidden,
+            "cannot extend sector's ({}) proof validity beyond {} epochs in the future (would be {})",
+            sector.sector_number,
+            policy.max_proof_validity,
+            new_expiration
+        ));
+    }
+
+    let mut new_sector = sector.clone();
+    new_sector.proof_expiration = new_expiration;
+    Ok(new_sector)
+}
+
+fn extend_commitment<BS, RT>(
+    rt: &RT,
+    decl: &ValidatedExpirationExtension,
+    sector: &SectorOnChainInfo,
+) -> Result<SectorOnChainInfo, ActorError>
+where
+    BS: Blockstore,
+    RT: Runtime<BS>,
+{
+    let curr_epoch = rt.curr_epoch();
+    if !can_extend_seal_proof_type(sector.seal_proof) {
+        return Err(actor_error!(
+            forbidden,
+            "cannot extend expiration for sector {} with unsupported \
```

Comment:
```suggestion
            "cannot extend commitment for sector {} with unsupported \
```
Hmm, I guess it makes sense to prevent this, but a bit less than it used to.